### DISCUSSION TITLE
Artifacthub with rds

### DISCUSSION THREAD

**Discussion by rleffd** [artifacthub/hub#4103] [General]
Hello,

I'm trying to setup artifacthub with an RDS (16.4) postgres instance. The master user is not used to deploy artifacthub. I have created a new user that have all rights on artifacthub databse.

But this user cannot create the pg_partman extension. This extension is available on my rds instance but the user cannot create the extension.

If the master user create the extension for my app user, the db-migrator-install says that extension already exists.

`ERROR: extension "pg_partman" already exists (SQLSTATE 42710)`

I wouldn't give the master user of my rds to the app. Is there a way to achieve that plz ?

**Comment by tegioz** [comment]
Hi @rleffd 👋

In:

https://github.com/artifacthub/hub/blob/52d39991046b21bb2e20b63171edc2e870efeaef/database/migrations/schema/034_partman.sql#L2

We should have:

`create extension if not exists pg_partman schema partman;`

That way you wouldn't get the error when the extension is created using the master user (*extension "pg_partman" already exists*), as we wouldn't try to create it again. We'll update this migration file to include this change. In the meantime, you could build your own `db-migrator` image with this fix if you'd like 🙂

**Reply by rleffd** [reply]
awesome !! I'll build a new image file and wait for the official update.

**Reply by tegioz** [reply]
Fixed by #4104 🙂
